src/main/java/htsjdk/samtools/SAMReadGroupRecord.java

```diff
@@ -90,6 +93,36 @@ public SAMReadGroupRecord(final String id, final SAMReadGroupRecord srcProgramRe
     public String getPlatform() { return getAttribute(PLATFORM_TAG); }
     public void setPlatform(final String platform) { setAttribute(PLATFORM_TAG, platform); }
 
+    /**
+     * @return the List of barcodes associated with this read group or null
+     */
+    public List<String> getBarcodes() {
+        final String barcodeString = getAttribute(BARCODE_TAG);
+        if(barcodeString == null){
+            return null;
+        } else if( barcodeString.isEmpty()){
+            return Collections.emptyList();
+        } else {
+            return Arrays.asList( barcodeString.split(BARCODE_SEPARATOR));
+        }
+    }
+
+    /**
+     * Set the barcodes associate with this ReadGroup.
+     * Note that null is treated as unsetting the attribute while an empty list is treated as a value set to have no barcodes.
+     * @param barcodes a list of barcodes to associate with this read group
+     */
+    public void setBarcodes(List<String> barcodes) {
```

Comment:
Same as before, a method to set with a  different separator might be useful for other implementations...

Reply:
```suggestion
    public void setBarcodes(final List<String> barcodes) {
```